charts/partials/templates/_proxy.tpl

```diff
@@ -0,0 +1,83 @@
+{{ define "partials.proxy" -}}
+- env:
+  - name: LINKERD2_PROXY_LOG
+    value: {{.LogLevel}}
+  - name: LINKERD2_PROXY_DESTINATION_SVC_ADDR
+    value: {{ternary "localhost.:8086" (printf "linkerd-destination.%s.svc.%s:8086" .ControlPlaneNamespace .ClusterDomain) (eq .Component "controller")}}
+  - name: LINKERD2_PROXY_CONTROL_LISTEN_ADDR
+    value: 0.0.0.0:{{.Port.Control}}
+  - name: LINKERD2_PROXY_ADMIN_LISTEN_ADDR
+    value: 0.0.0.0:{{.Port.Admin}}
+  - name: LINKERD2_PROXY_OUTBOUND_LISTEN_ADDR
+    value: 127.0.0.1:{{.Port.Outbound}}
+  - name: LINKERD2_PROXY_INBOUND_LISTEN_ADDR
+    value: 0.0.0.0:{{.Port.Inbound}}
+  - name: LINKERD2_PROXY_DESTINATION_PROFILE_SUFFIXES
+    {{- $internalProfileSuffix := printf "svc.%s" .ClusterDomain }}
+    value: {{ternary "." $internalProfileSuffix .EnableExternalProfile}}
+  - name: LINKERD2_PROXY_INBOUND_ACCEPT_KEEPALIVE
+    value: {{.InboundAcceptKeepAlive}}
+  - name: LINKERD2_PROXY_OUTBOUND_CONNECT_KEEPALIVE
+    value: {{.OutboundAcceptKeepAlive}}
+  - name: _pod_ns
+    valueFrom:
+      fieldRef:
+        apiVersion: v1
+        fieldPath: metadata.namespace
+  - name: LINKERD2_PROXY_DESTINATION_CONTEXT
+    value: ns:$(_pod_ns)
+  - name: LINKERD2_PROXY_IDENTITY_DIR
+    value: /var/run/linkerd/identity/end-entity
+  - name: LINKERD2_PROXY_IDENTITY_TRUST_ANCHORS
+    value: |
+    {{- .IdentityTrustAnchors | trim | nindent 6 }}
+  - name: LINKERD2_PROXY_IDENTITY_TOKEN_FILE
+    value: /var/run/secrets/kubernetes.io/serviceaccount/token
+  - name: LINKERD2_PROXY_IDENTITY_SVC_ADDR
+    value: linkerd-identity.{{.ControlPlaneNamespace}}.svc.cluster.local:8080
+  - name: _pod_sa
+    valueFrom:
+      fieldRef:
+        apiVersion: v1
+        fieldPath: spec.serviceAccountName
+  - name: _l5d_ns
+    value: {{.ControlPlaneNamespace}}
+  - name: _l5d_trustdomain
+    value: {{.ClusterDomain}}
```

Comment:
Currently in the code this value is being set to what gets passed in `--identity-trust-domain` (that defaults to `cluster.local`). I understand #3148 will allow to set cluster domain, but I think this one can still be different.